scheduling-manager/src/test/java/com/amazonaws/blox/scheduling/SchedulingSystemTest.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.scheduling;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.amazonaws.blox.dataservicemodel.v1.client.DataService;
+import com.amazonaws.blox.dataservicemodel.v1.model.DeploymentMethod;
+import com.amazonaws.blox.dataservicemodel.v1.model.EnvironmentRevision;
+import com.amazonaws.blox.dataservicemodel.v1.model.operations.DescribeEnvironmentRevisionResponse;
+import com.amazonaws.blox.dataservicemodel.v1.model.operations.DescribeTargetEnvironmentRevisionRequest;
+import com.amazonaws.blox.dataservicemodel.v1.model.operations.ListClustersResponse;
+import com.amazonaws.blox.dataservicemodel.v1.model.operations.ListEnvironmentsRequest;
+import com.amazonaws.blox.dataservicemodel.v1.model.operations.ListEnvironmentsResponse;
+import com.amazonaws.blox.lambda.TestLambdaFunction;
+import com.amazonaws.blox.scheduling.manager.ManagerHandler;
+import com.amazonaws.blox.scheduling.manager.ManagerInput;
+import com.amazonaws.blox.scheduling.manager.ManagerOutput;
+import com.amazonaws.blox.scheduling.reconciler.CloudWatchEvent;
+import com.amazonaws.blox.scheduling.reconciler.ReconcilerHandler;
+import com.amazonaws.blox.scheduling.scheduler.SchedulerHandler;
+import com.amazonaws.blox.scheduling.scheduler.SchedulerInput;
+import com.amazonaws.blox.scheduling.scheduler.SchedulerOutput;
+import com.amazonaws.blox.scheduling.scheduler.engine.SchedulerFactory;
+import com.amazonaws.blox.scheduling.state.ClusterSnapshot;
+import com.amazonaws.blox.scheduling.state.ClusterSnapshot.ContainerInstance;
+import com.amazonaws.blox.scheduling.state.ECSState;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.concurrent.CompletableFuture;
+import lombok.extern.log4j.Log4j2;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import software.amazon.awssdk.services.ecs.ECSAsyncClient;
+import software.amazon.awssdk.services.ecs.model.StartTaskRequest;
+import software.amazon.awssdk.services.ecs.model.StartTaskResponse;
+
+@Log4j2
+public class SchedulingSystemTest {
+
+  private static final String CLUSTER_ARN = "arn:::::cluster1";
```

Comment:
we're going to have to add validation for inputs so these will fail

Reply:
Those are actually valid ARNs according to the spec. If we want to really validate inputs, we'll need a robust ARN parser.

I have some ideas for that I'd like to talk through, mind if we come back to that?

Reply:
sure